src/tests/GfxReplayTest.cpp

```diff
@@ -449,6 +456,101 @@ void GfxReplayTest::testSimulatorIntegration() {
   }
 }
 
+// test lights data by recording and playback through the simulator interface
+void GfxReplayTest::testLightIntegration() {
+  const auto compareLights = [&](const LightInfo& a, const LightInfo& b) {
+    CORRADE_COMPARE(a.vector, b.vector);
+    CORRADE_COMPARE(static_cast<int>(a.model), static_cast<int>(b.model));
+    CORRADE_COMPARE(a.color, b.color);
+  };
+
+  const auto compareLightSetups = [&](const LightSetup& a,
+                                      const LightSetup& b) {
+    CORRADE_COMPARE(a.size(), b.size());
+    for (int i = 0; i < a.size(); ++i) {
+      compareLights(a[i], b[i]);
+    }
+  };
+
+  const auto testFilepath =
+      Corrade::Utility::Path::join(DATA_DIR, "./gfx_replay_test.json");
+  const LightInfo pointLight0{Magnum::Vector4{1.5f, 2.0f, 2.5f, 1.0f},
+                              {1.5, 2.0, 5.0},
+                              LightPositionModel::Global};
+  const LightInfo pointLight1{Magnum::Vector4{-10.0f, 4.25f, 10.0f, 1.0f},
+                              {5.0, 5.0, 0.0},
+                              LightPositionModel::Global};
+  const LightInfo pointLight2{Magnum::Vector4{0.0f, 1.2f, -4.0f, 1.0f},
+                              {4.0, 4.0, 4.0},
+                              LightPositionModel::Global};
+  const LightInfo pointLight3{Magnum::Vector4{0.1f, 0.2f, -0.3f, 1.0f},
+                              {0.0, 0.0, 1.0},
+                              LightPositionModel::Global};
+  const LightSetup lightSetup0{pointLight0, pointLight1};
+  const LightSetup lightSetup1{pointLight2};
+  const LightSetup lightSetup2{pointLight3};
+  const std::string lightSetup0Key = "custom_lighting_0";
```

Comment:
Ah, another thing, let's only use the default light setup key. I believe the key is `""`, or omit the key argument altogether in the API calls. I discussed this in the design doc.

The problem with using three different light setup keys is that we never actually change/delete lights unless we repeatedly call `setLightSetup` for the same key.